src/lib/helpers.ts

```diff
@@ -220,6 +220,53 @@ export function getOvalConfigs(input: string): OvalConfigs {
   throw new Error(`Value "${input}" is valid JSON but is not OvalConfigs records`);
 }
 
+// Get OvalHeaderConfigs from the header string or throw an error if the header is invalid.
+export const getOvalHeaderConfigs = (
+  header: string | string[] | undefined,
+  ovalConfigs: OvalConfigs,
+): { valid: boolean; ovalHeaderConfigs: OvalHeaderConfigs | undefined } => {
+  if (!header) return { valid: true, ovalHeaderConfigs: undefined };
+  if (typeof header !== "string") return { valid: false, ovalHeaderConfigs: undefined };
+
+  let ovalHeaderConfigs: OvalHeaderConfigs;
+  let valid = true;
+  try {
+    ovalHeaderConfigs = JSON.parse(header);
+    if (!ovalHeaderConfigs.unlockAddresses)
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+    if (!Array.isArray(ovalHeaderConfigs.unlockAddresses))
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+    if (ovalHeaderConfigs.unlockAddresses.find((ovalHeaderConfig) => !isAddress(ovalHeaderConfig.ovalAddress)))
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
+    if (ovalHeaderConfigs.unlockAddresses.find((ovalHeaderConfig) => !ovalConfigs[ovalHeaderConfig.ovalAddress]))
+      throw new Error(`Value "${header}" cannot be converted to OvalConfigs records`);
```

Comment:
Since we have branches for each of the different types of errors, thoughts on making them more specific?

"Found unlocker address in header that is not a configured oval instance" as an example.